tests/components/plugwise/test_init.py

```diff
@@ -0,0 +1,44 @@
+"""Tests for the Plugwise Climate integration."""
+
+import asyncio
+
+from Plugwise_Smile.Smile import Smile
+
+from tests.components.plugwise.common import async_init_integration
+
+
+async def test_smile_unauthorized(hass, mock_smile_unauth):
+    """Test failing unauthorization by Smile."""
+    assert await async_init_integration(hass, mock_smile_unauth) is False
+
+
+async def test_smile_error(hass, mock_smile_error):
+    """Test server error handling by Smile."""
+    assert await async_init_integration(hass, mock_smile_error) is False
+
+
+async def test_smile_notconnect(hass, mock_smile_notconnect):
+    """Connection failure error handling by Smile."""
+    mock_smile_notconnect.connect.return_value = False
+    try:
+        assert await async_init_integration(hass, mock_smile_notconnect) is False
+    except Smile.PlugwiseError as exception:
```

Comment:
There should be no exceptions leaking out from the integration. Where does this happen?

Reply:
`Smile.PlugwiseError` is a module exception - I could be more specific indicating `except Smile.InvalidAuthentication` but how would I assert it then? (Or I'm missing what you're pointing me towards.

Reply:
What is raising this error?

Reply:
They way I wrote it was to test if the correct exception is raised I think I got that idea from `august` or likewise not sure. It works by leaving them out just asserting that 'it wont work' - it's just that I don't test for specifics (next commit is like that).

Reply:
Since the error never leaves the integration we can't test it by catching the exception. We have to assert the state of the config entry. Eg if we expect a retry the state should be `ENTRY_STATE_SETUP_RETRY`.

https://github.com/home-assistant/core/blob/cdc93d7110fac0d8d1a5aaae1d38a287024f5654/homeassistant/config_entries.py#L52-L63

Reply:
I think I rewrote all of these (positive and negative) now